confignode/src/main/java/org/apache/iotdb/confignode/procedure/env/DataNodeRemoveHandler.java

```diff
@@ -155,6 +157,39 @@ public TDataNodeLocation findDestDataNode(TConsensusGroupId regionId) {
     return newNode.get();
   }
 
+  public TSStatus setRegionStatus(TConsensusGroupId regionId, TDataNodeLocation originalDataNode) {
+    TSetRegionStatusReq req = new TSetRegionStatusReq(regionId, RegionStatus.Removing.getStatus());
+
+    TSStatus status =
+        configManager.getNodeManager().getNodeStatusByNodeId(originalDataNode.getDataNodeId())
+                == NodeStatus.Unknown
+            ? SyncDataNodeClientPool.getInstance()
+                .sendSyncRequestToDataNodeWithGivenRetry(
+                    originalDataNode.getInternalEndPoint(),
+                    req,
+                    DataNodeRequestType.SET_REGION_STATUS,
+                    1)
+            : SyncDataNodeClientPool.getInstance()
+                .sendSyncRequestToDataNodeWithRetry(
+                    originalDataNode.getInternalEndPoint(),
+                    req,
+                    DataNodeRequestType.SET_REGION_STATUS);
```

Comment:
I notice that this interface is only used in RegionMigrateProcedure. However, is it safe if we migrate a Region that located at a Unknown DataNode? If not, we can just refuse this migration but not try to modify the RegionStatus with 1 retry.

Reply:
Because a region migration may also be caused by a DataNode removement, which is currently supported on unknown DataNode.